### PR TITLE
Add outOfView comp

### PR DESCRIPTION
**Pull request by slmjkdbtl**
As discussed in #422 , would be nice to have a component for general out-of-view actions, including hiding, pausing, destroying, or custom event etc.

```js
const characterFarAway = add([
    pos(1200, 80),
    outOfView({ hide: true, pause: true }),
])
```

and `cleanup()` component can be just a thin wrapper of `outOfView({ destory: true })`

`outOfView` is slightly verbose comparing to other built-in comps, is there a more succinct name?

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -3199,6 +3217,67 @@ export interface FollowComp extends Comp {
...
+	 */
+	pause?: boolean,
+	/**
+	 * If destroy object when out of view.
+	 */
+	destroy?: boolean,
+	/**
+	 * The screen bound offset.
+	 */
+	offset?: Vec2,
```
Comment (slmjkdbtl) [inline src/types.ts:3236]: Not sure about this name, maybe call it `margin`?

**Comment by lajbel** [comment]
suggestion: rename to out?

**Comment by slmjkdbtl** [comment]
@lajbel An option, I wonder if people won't understand what it does from the name

**Comment by lajbel** [comment]
> @lajbel An option, I wonder if people won't understand what it does from the name

outview()?

File: src/types.ts (hunk 1 above)
Comment (lajbel) [reply src/types.ts]: nop, offset is better (my oppinion)

**Comment by slmjkdbtl** [comment]
> > @lajbel An option, I wonder if people won't understand what it does from the name
> 
> outview()?

yeah i kinda like that more than `outOfView`

**Review by lajbel — APPROVED** [review]
cool name

**Pull request merged**